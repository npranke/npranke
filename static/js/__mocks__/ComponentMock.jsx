function ComponentMock(props) {
    return (
        <div className="component-mock" role="document">
            {
                `${Object.keys(props).toString()} `
                + `${Object.values(props).toString()}`
            }
        </div>
    )
}

export default ComponentMock
