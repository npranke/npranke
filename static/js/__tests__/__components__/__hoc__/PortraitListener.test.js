import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { mount } from 'enzyme'
import React from 'react'

import ComponentMock from '@__mocks__/ComponentMock'

import withPortraitListener from '@components/hoc/PortraitListener'

Enzyme.configure({ adapter: new Adapter() })

describe('PortraitListener', () => {
    beforeEach(() => {
        window.matchMedia = jest.fn().mockReturnValue({
            matches: false,
            addListener: jest.fn(),
            removeListener: jest.fn(),
        })
    })

    test('wrapped component has isPortrait prop', () => {
        const WrappedComponent = withPortraitListener(ComponentMock)

        const wrappedComponent = mount(<WrappedComponent />)

        expect(
            wrappedComponent.find('ComponentMock').props().isPortrait,
        ).toBeDefined()
    })
})

describe('PortraitListener snapshot', () => {
    test('matches snapshot', () => {
        const WrappedComponent = withPortraitListener(ComponentMock)

        const wrappedComponent = mount(<WrappedComponent />)

        expect(wrappedComponent).toMatchSnapshot()
    })
})
