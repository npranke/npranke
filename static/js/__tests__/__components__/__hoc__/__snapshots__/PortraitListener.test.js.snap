// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortraitListener snapshot matches snapshot 1`] = `
<Component>
  <ComponentMock
    isPortrait={false}
  >
    <div
      className="component-mock"
    >
      mock
    </div>
  </ComponentMock>
</Component>
`;
